tests/conftest.py

```diff
@@ -725,6 +732,155 @@ def _launch_image(image: DockerImage, timeout: float = 2.0, max_attempts: int =
             container.remove()
 
 
+def _pre_launch(image: DockerImage) -> None:
+    """Run pre-launch checks."""
+    image.check_skip()
+    image.stop_if_already_running()
+
+
+def _start_container(
+    image: DockerImage, container, timeout: float, max_attempts: int
+) -> None:
+    """
+    Start a container.
+
+    :param image: an instance of Docker image.
+    :param container: the container to start, created from the image.
+    :param timeout: timeout to launch
+    :param max_attempts: max launch attempts
+    """
+    container.start()
+    logger.info(f"Setting up image {image.tag}...")
+    success = image.wait(max_attempts, timeout)
+    if not success:
+        container.stop()
+        logger.error(
+            f"{DOCKER_PRINT_SEPARATOR}Logs from container {container.name}:\n{container.logs().decode()}"
+        )
+        container.remove()
+        pytest.fail(f"{image.tag} doesn't work. Exiting...")
+    else:
+        logger.info("Done!")
+        time.sleep(timeout)
+
+
+def _stop_container(container: Container, tag: str) -> None:
+    """Stop a container."""
+    logger.info(f"Stopping container {container.name} from image {tag}...")
+    container.stop()
+    try:
+        logger.info(
+            f"{DOCKER_PRINT_SEPARATOR}Logs from container {container.name}:\n{container.logs().decode()}"
+        )
+        if str(container.name).startswith("node"):
+            logger.info(
+                f"{DOCKER_PRINT_SEPARATOR}Logs from container log file {container.name}:\n"
+            )
+            bits, _ = container.get_archive(f"/logs/{container.name}.txt")
+            for chunk in bits:
+                logger.info(chunk.decode())
+    except (ImageNotFound, NotFound) as e:
+        logger.error(e)
+    finally:
+        container.remove()
+
+
+def launch_many_containers(
+    image: DockerImage, timeout: float = 2.0, max_attempts: int = 10
+) -> Generator[DockerImage, None, None]:
+    """Launch many containers from an image."""
+    _pre_launch(image)
+    containers = image.create()
+    for container in containers:
+        _start_container(image, container, timeout, max_attempts)
+    yield image
+    for container in containers:
+        _stop_container(container, image.tag)
+
+
+LOCAL_ADDRESS = "0.0.0.0"
+ACN_CONFIGURATION: Dict[str, str] = dict(
+    AEA_P2P_ID="54562eb807d2f80df8151db0a394cac72e16435a5f64275c277cae70308e8b24",
+    AEA_P2P_URI_PUBLIC=f"{LOCAL_ADDRESS}:5000",
+    AEA_P2P_URI=f"{LOCAL_ADDRESS}:5000",
+    AEA_P2P_DELEGATE_URI=f"{LOCAL_ADDRESS}:11000",
+    AEA_P2P_URI_MONITORING=f"{LOCAL_ADDRESS}:8080",
+    ACN_LOG_FILE="/acn/libp2p_node.log",
+)
```

Comment:
not relevant atm, I used this for `UseACNNode` - the single container I started out with to setup the ACN with a bootstrap node and two entry nodes (`UseACNWithBootstrappedEntryNodes` below) .